pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -126,18 +97,76 @@ def _update_next_infection_status(self, person: Person):
+                                     'probabilities')
+
+            next_infection_status_number = random.choices(outcomes, weights)[0]
+            next_infection_status =\
+                InfectionStatus(next_infection_status_number)
+            person.next_infection_status = next_infection_status
+
+    def _update_time_status_change(self, person, time):
+        """Calculates transition time as calculated in CovidSim,
+        and updates the time_of_status_change for the given
+        Person, given as the time until next infection status
+        for a person who has a new infection status. If it is expected that
+        the person will not transition again (for example in Recovered or Dead
+        statuses), then the time of status change is set to infinity.
 
-        row_index = person.infection_status.name
-        weights = self.state_transition_matrix.loc[row_index].to_numpy()
-        outcomes = range(1, self.number_of_states + 1)
-
-        if len(weights) != len(outcomes):
-            raise AssertionError('The number of infection statuses must \
-                                match the number of transition probabilities')
+        Parameters
+        ----------
+        Person : Person
+            Instance of Person class with :class:`InfectionStatus` attributes
+        time : float
+            Current simulation time
 
-        next_infection_status_number = random.choices(outcomes, weights)[0]
-        next_infection_status = InfectionStatus(next_infection_status_number)
-        person.next_infection_status = next_infection_status
+        """
+        # Defines the transition time. If the person will not transition again,
+        # the transition time is set to infinity. Else, the transition time is
+        # defined using the TransitionTimeMatrix class, with the method
+        # `choose` from the InverseCdf class.
+        if (person.infection_status == InfectionStatus.Recovered or
+                person.infection_status == InfectionStatus.Dead):
+            transition_time = np.inf
+        else:
+            row_index = person.infection_status.name
+            column_index = person.next_infection_status.name
+            transition_time_icdf_object =\
+                self.transition_time_matrix.loc[row_index, column_index]
+            # Checks for susceptible to exposed case
+            # where transition time is zero
+            try:
+                transition_time =\
+                    transition_time_icdf_object.icdf_choose_noexp()
+            except AttributeError as e:
+                if "object has no attribute 'icdf_choose_noexp'" in str(e):
+                    transition_time = transition_time_icdf_object
+                    assert isinstance(
+                        transition_time_icdf_object,
+                        (float, int)), \
+                        ("Entries of transition time matrix \
+                        must either be ICDF" + " objects or numbers")
```

Comment:
```suggestion
                        (float, int)), \
                        ("Entries of transition time matrix"
                         + " must either be ICDF objects or numbers")
```
[PEP8](https://peps.python.org/pep-0008/#maximum-line-length) is really quite against backslashes so we should avoid them if possible, and they're definitely not needed here as you can break the comment across lines.